code-challenges/CC9_reduce/challenges-09.test.js

```diff
@@ -0,0 +1,299 @@
+
+const calculateAverage = (arr) => {
+  // Solution code here...
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 6
+
+Write a function named countPrimeNumbers that, given an array elements as input, uses reduce to count the number of elements that are prime numbers.
+
+You are welcome to use the provided isPrime function.
+------------------------------------------------------------------------------------------------ */
+
+const isPrime = (value) => {
+  for (let i = 2; i < value; i++) {
+    if (value % i === 0) {
+      return false;
+    }
+  }
+  return value > 1;
+};
+
+const countPrimeNumbers = (arr) => {
+  // Solution code here...
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 7 - Stretch Goal
+
+Write a function named extractState that, given the snorlaxData, below, uses reduce to return the object whose 'name' property matches the given string.
+
+If the input array does not have a stat with that specific name, the function should return null.
+------------------------------------------------------------------------------------------------ */
+
+const snorlaxData = {
+  stats: [
+    {
+      stat: {
+        url: 'https://pokeapi.co/api/v2/stat/6/',
+        name: 'speed',
+      },
+      effort: 5,
+      baseStat: 30,
+    },
+    {
+      stat: {
+        url: 'https://pokeapi.co/api/v2/stat/5/',
+        name: 'special-defense',
+      },
+      effort: 2,
+      baseStat: 110,
+    },
+    {
+      stat: {
+        url: 'https://pokeapi.co/api/v2/stat/4/',
+        name: 'special-attack',
+      },
+      effort: 9,
+      baseStat: 65,
+    },
+  ],
+  name: 'snorlax',
+  weight: 4600,
+};
+
+const extractStat = (statName, arr) => {
+  // Solution code here...
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 8 - Stretch Goal
+
+Write a function named extractChildren that, given the array of characters from challenge 4, accomplishes the following:
+
+1) Uses filter to return an array of the characters that contain the letter 'a' in their name
+
+2) Then, uses reduce to return an array of all the children's names in the filtered array
+------------------------------------------------------------------------------------------------ */
+
+const extractChildren = (arr) => {
+  // Solution code here...
+};
+
+/* ------------------------------------------------------------------------------------------------
+TESTS
+
+All the code below will verify that your functions are working to solve the challenges.
+
+DO NOT CHANGE any of the below code.
+
+Run your tests from the console: jest challenges-09.test.js
+------------------------------------------------------------------------------------------------ */
+
+x('Testing challenge 1', () => {
```

Comment:
This seems like a typo. It should be xdescribe if you wanted to skip these tests.